### DISCUSSION TITLE
protoc-gen-entgrpc: mutating methods should regard null/zero values and field nullability

### DISCUSSION THREAD

**Discussion by rotemtam** [ent/ent#1350] [Ideas]
# Summary 💡
Currently `protoc-gen-entgrpc` does not consider zero/nil values in the generated `Create` and `Update` methods. 
For example, the `Create` method will Set all available fields:
```go
res, err := svc.client.User.Create().
  SetExp(uint64(user.GetExp())).
  SetJoined(entproto.ExtractTime(user.GetJoined())).
  SetPoints(uint(user.GetPoints())).
  SetStatus(toEntUser_Status(user.GetStatus())).
  SetUserName(string(user.GetUserName())).
  SetGroupID(int(user.GetGroup().GetId())).
  Save(ctx)
```
If for example, the `Joined` field has a default value, the user has no way to pass a nil/zero value to let ent use the default for the field.

In addition, if the calling user does not set a Group with a valid ID, the generated code will execute .SetGroupId(0), which will cause the insert to fail on a foreign key constraint, since no Group with ID zero exists.

**Comment by rotemtam** [comment]
My initial thought on this is to use proto 3 [Wrapper types](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/wrappers.proto) instead of primitives for fields that are Nillable and then to check for presence before invoking the ent `Set<Field>`.

WDYT @Bladrak @tmc @godcong ?

**Reply by Bladrak** [reply]
Looks like a good insight indeed 👍
